tidy3d/components/grid/grid_spec.py

```diff
@@ -488,6 +510,19 @@ def make_grid(
             wavelength = self.wavelength_from_sources(sources)
             log.info(f"Auto meshing using wavelength {wavelength:1.4f} defined from sources.")
 
+        # Warn user if `GridType` along some axis is not `AutoGrid` and
+        # override_structures is not empty. The override structures
+        # are not effective along those axes.
+        for axis_ind, override_used_axis, grid_axis in zip(
+            ["x", "y", "z"], self.override_structures_used, [self.grid_x, self.grid_y, self.grid_z]
+        ):
+            if override_used_axis and not isinstance(grid_axis, AutoGrid):
+                log.warning(
+                    f"Override structures take no effect along {axis_ind}-axis. "
+                    f"If intending to apply override structures to this axis, "
+                    "use `AutoGrid`."
```

Comment:
`'AutoGrid'`. We use `'variable'` in error/warning messages,
```
``variable``
```
in docstrings/field descriptions (RST format), and
```
`variable`
```
in markdown files like README.md and notebooks.

Reply:
It has been updated. thanks!